src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControl.java

```diff
@@ -212,7 +212,7 @@ private static Pattern createPatternForExactMatch(String expression) {
      * @param rule the rule to be added.
      */
     protected void addImportRule(AbstractImportRule rule) {
-        rules.addFirst(rule);
+        rules.addLast(rule);
```

Comment:
0)Before we merge this can you reckeck history of this line.
Such code usually written on purpose.

1)This line mean that enforcing sequence is the most last rule wins.
So it is  a bit oposite to what you expect, but still make sense.
Please recheck one more time.

2) we need to document sequence of applying in xdoc

3) we need example in xdoc, as it is far from abvioous,

Reply:
> 0)Before we merge this can you reckeck history of this line.

The code has been here since the beginning.
https://github.com/checkstyle/checkstyle/commit/1951594e90a04aa2f2b14021ce3318604218ba19#diff-7749ecc84be7ad0bb709565b05a8e0ccR75

Original XDoc did not specify anything.

Test setup does write classes in reverse alphabetical order, but it is not fully clear if it is done on purpose.
https://github.com/checkstyle/checkstyle/commit/1951594e90a04aa2f2b14021ce3318604218ba19#diff-bd322f2d6a0a3503dee0f734956cd24fR16

> This line mean that enforcing sequence is the most last rule wins.

Incorrect. We are still enforcing first rule wins, we are just not reversing the order of the rules.
Look at my UT as an example. If last rule wins was the case, it would have allowed the import and printed no violation as the package is allowed.

Reply:
@romani 
> we need to document sequence of applying in xdoc
we need example in xdoc, as it is far from abvioous,

Done.